WordPress/gutenberg
# Add filter to turn off Interactivity API for a block

## What?


Add a `gutenberg_should_block_use_interactivity_api` filter which allows the experimental Interactivity API to be disabled for a given block.

## Why?


This is a follow-up to https://github.com/WordPress/gutenberg/pull/52553 for reasons explained in https://github.com/WordPress/gutenberg/pull/50906#issuecomment-1632902880. Currently the only way to prevent the Interactivity API from being used for the File block or Navigation block is to hack the `GUTENBERG_IS_PLUGIN` constant to be `false`. Since the experimental Interactivity API is not behind an experiment feature flag, this filter is needed in order to be able to access and hack on the old implementations.

## How?


The `gutenberg_should_block_use_interactivity_api` filter is added in a function of the same name. This is used in places where the `GUTENBERG_IS_PLUGIN` constant was used to determine whether to use the Interactivity API.

## Testing Instructions

Add a mu-plugin that that includes this code:

```php
add_filter( 'gutenberg_should_block_use_interactivity_api', '__return_false' );
```

When absent, a post with the File block and Navigation block contain in the footer:

```html
<script src='http://localhost:8888/wp-content/plugins/gutenberg/build/interactivity/index.min.js?ver=1689183370' id='wp-interactivity-js'></script>
<script src='http://localhost:8888/wp-content/plugins/gutenberg/build/block-library/blocks/navigation/view-interactivity.min.js?ver=3dae10865f3f5dd45c1d' id='wp-block-navigation-view-js'></script>
<script src='http://localhost:8888/wp-content/plugins/gutenberg/build/block-library/blocks/file/view-interactivity.min.js?ver=033f781a5a681b19ffeb' id='wp-block-file-view-js'></script>
```

When present, those scripts do not appear in the footer and instead these scripts appear in the `head`:

```html
<script src='http://localhost:8888/wp-content/plugins/gutenberg/build/block-library/blocks/navigation/view.min.js?ver=7e248642387235da6fc6' id='wp-block-navigation-view-js'></script>
<script src='http://localhost:8888/wp-content/plugins/gutenberg/build/block-library/blocks/navigation/view-modal.min.js?ver=bc9582176e71f4196e97' id='wp-block-navigation-view-2-js'></script>
<script src='http://localhost:8888/wp-content/plugins/gutenberg/build/block-library/blocks/file/view.min.js?ver=7a5cb419b4808ec7a475' id='wp-block-file-view-js'></script>
```
